### PR TITLE
Implement Health Check Status

### PR DESCRIPTION
**Pull request by ademar111190**
Implement Health Check Status
Fixes https://github.com/breez/c-breez/issues/729

# How it looks like

|Maintanance|Service disruption|
|:-:|:-:|
|maintanance|disruption|

# How to test

## Method 1 (changing the code)
- Change the code to emit a different `HealthCheckStatus` than `Operational`
- You should see at the home page a flushbar like that one in screenshot

## Method 2 (being patient)
- Wait for a moment in the future when breez service is in maintenance or has it service disrupted
- You should see at the home page a flushbar like that one in screenshot

### CODE REVIEW

**Comment by erdemyerebasmaz** [comment]
@roeierez  @dangeross Is it possible to convert this function into a stream? It seems more intuitive than setting up a polling mechanism.

File: lib/handlers/health_check_handler.dart [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      return;
+    }
+
+    final texts = context.texts();
+    final themeData = Theme.of(context);
+    final bloc = context.read<HealthCheckBloc>();
+
+    _listener?.cancel();
+    _listener = bloc.stream.listen((event) {
+      if (event == HealthCheckStatus.Maintenance || event == HealthCheckStatus.ServiceDisruption) {
```
Comment (roeierez) [inline lib/handlers/health_check_handler.dart:35]: Shouldn't we also remove the flushbar when the helth check is normal again?

**Comment by roeierez** [comment]
> @roeierez @dangeross Is it possible to convert `serviceHealthCheck` function into a stream? It seems more intuitive than setting up a polling mechanism.

It means the SDK either needs to poll or maintain a live connection to the breez server. I think both are overkill.
IMO we don't need to poll here wither, certainly not every 30 seconds. Perhaps only check on restart the app? What are we doing in breez mobile?
@ademar111190 what do you think about adding a button on the flashbar to re-check the status?

**Review by roeierez — CHANGES_REQUESTED** [review]

**Comment by ademar111190** [comment]
@roeierez @erdemyerebasmaz  breez mobile pulls every 30 seconds; I copied the logic from there. Honestly, I believe that checking it one time once the app opens is enough, but maybe I am missing something that was discussed at the time people built it on breez mobile.

Should I change it to check at the time the app opens only? If yes, should I change it for breez mobile too?

File: lib/handlers/health_check_handler.dart (hunk 1 above)
Comment (ademar111190) [reply lib/handlers/health_check_handler.dart:35]: good catch; fixed.

**Comment by dangeross** [comment]
Just as a thought - maybe when a different request to the SDK fails, we can poll the health status?

**Comment by roeierez** [comment]
> Just as a thought - maybe when a different request to the SDK fails, we can poll the health status? 

In theory you are right @dangeross but in practice I think there are many edge cases such as API to the lsp and server works perfectly but the lsp node is down for maintenance so I suspect it doesn't worth the effort.

**Comment by roeierez** [comment]
> @roeierez @erdemyerebasmaz  breez mobile pulls every 30 seconds; I copied the logic from there. Honestly, I believe that checking it one time once the app opens is enough, but maybe I am missing something that was discussed at the time people built it on breez mobile.
> 
> Should I change it to check at the time the app opens only? If yes, should I change it for breez mobile too?

No need to change breez mobile.
How about one time check on start and poll every minute if the status is not normal? Once normal remove the flushbar.
Also IMO good to have a button in the flashbar to refresh the status manually.

**Comment by ademar111190** [comment]
Ok; not changing breez mobile.

I did apply your suggestions @roeierez also I moved the call trigger to a separate method; so if we want to trigger it again after some fail or other flow as @dangeross has suggested, we just need to get the bloc and trigger the call.

**Review by roeierez — APPROVED** [review]
LGTM

**Pull request merged**